frontend/src/components/NavBar.js

```diff
@@ -37,7 +44,12 @@ const NavBar = () => {
   }
 
   const _renderAccountIcon = () => {
-    if (!isUserLoggedIn()) {
+    if (
+      !isUserLoggedIn() &&
+      (location.pathname === baseUrl ||
+        location.pathname === signupUrl ||
+        location.pathname === loginUrl)
```

Comment:
Added the pathname checks so that the account icon will only disappear on these pages